src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class FigureSupplier implements Figure, Draw {
+
+    public int getDefaultRadius() {
+        int radius;
+        radius = 10;
+        return radius;
+    }
+
+    public int getRandomFigure() {
+        //for the generation of figures with episodic powers
+        int randvalues = new Random().nextInt(4) + 1;
+        return randvalues;
+    }
+
+    public String getDefaultColor() {
+        String color;
+        color = "white";
+        return color;
+    }
```

Comment:
Methods and signature of your class not comply with it's purpose
There should be something like below
```suggestion
public class FigureSupplier {
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();

    public Figure getRandomFigure() {
        ...some code here
    }

    public Figure getDefaultFigure() {
        ...some code
    }
```